src/session.c

```diff
@@ -51,6 +59,60 @@ typedef struct SESSION_INSTANCE_TAG
 #define UNDERLYING_CONNECTION_NOT_OPEN 0
 #define UNDERLYING_CONNECTION_OPEN 1
 
+static void remove_link_endpoint(LINK_ENDPOINT_HANDLE link_endpoint)
+{
+    if (link_endpoint != NULL)
+    {
+        LINK_ENDPOINT_INSTANCE* endpoint_instance = (LINK_ENDPOINT_INSTANCE*)link_endpoint;
+        SESSION_INSTANCE* session_instance = endpoint_instance->session;
+        uint64_t i;
```

Comment:
>uint64_t i [](start = 8, length = 10)

Unnecessary type conversion.  i is an uint64_t and link_endpoint_count is a uint32_t.  Please make them the same type.

Reply:
Absolutely. i is now uint32_t

---
In reply to: [189095882](https://github.com/Azure/azure-uamqp-c/pull/236#discussion_r189095882) [](ancestors = 189095882)